### DISCUSSION TITLE
How to copy the type of when hovering over the property

### DISCUSSION THREAD

**Discussion by jcubic** [ananthakumaran/tide#471] [Q&A] [answered]
I just found that you can hover or click on a prop in React to get the proper type of prop. (learned while watching a video when someone used VS Code).
e.g. I have this React Native code:

```jsx
type VoidFunction = () => void;

type ButtonProps = {
    children: ReactNode;
    onPress?: VoidFunction;
};

const Button: FC<ButtonProps> = ({ children, onPress }) => {
    const style = ({pressed}: {pressed: boolean}) => {
        return pressed ? [styles.pressable, styles.pressed] : styles.pressable
    };
    return (
        <Pressable onPress={onPress} style={style}>
          <View style={styles.container}>
            <Text style={styles.text}>{ children }</Text>
          </View>
        </Pressable>
    );
};
```

and I have voidFunction for the type of the onPress handler. But when I hover or click on the prop I get the proper type for that event.

So the question is how I can copy that definition, it appears in the minibuffer but when I click on the minibuffer it disappears. When I have an error in TypeScript I often use click and move the keyboard so the error appears in *messages* buffer, and there was also some function from flyspell (I think) that showed all the errors, but that will not work with a type of the thing you hover over.

**Comment by ananthakumaran** [comment, marked as answer]
Check the faq in readme

**Reply by jcubic** [reply]
Thanks, one more question how are the docs for the prop added in typescript?

I got this:

```typescript
(property) PressableProps.onPress?: ((event: GestureResponderEvent) => void) | null | undefined

Called when a single tap gesture is detected.
```

**Reply by ananthakumaran** [reply]
This is handled by [tsserver](https://github.com/microsoft/TypeScript/wiki/Standalone-Server-(tsserver)) and typescript. You might have to ask at other places to get the details.

**Reply by jcubic** [reply]
ok, thanks
